nuheat/thermostat.py

```diff
@@ -338,4 +342,9 @@ def set_data(self, post_data):
         params = {
             "serialnumber": self.serial_number
         }
-        self._session.request(config.THERMOSTAT_URL, method="POST", data=post_data, params=params)
+        self._session.request(
+            url=get_thermostat_url(brand=self._session._brand),
```

Comment:
with my suggestion below, i believe this could be:

```suggestion
            url=self._url,
```